### PR TITLE
crypto: Add support for generic platcrypt crypto library

### PR DESCRIPTION
**Pull request by deebee-v2**
Allows for platform dependent implementations of exported crypto API

Signed-off-by: Darren Roche <darren.broche@gmail.com>

### CODE REVIEW

File: core/core.mk [hunk 1]
```diff
@@ -88,9 +91,17 @@ libdir = lib/libmpa
 include mk/lib.mk
 base-prefix :=
 
+ifeq ($(WITH_CRYPTO), tomcrypt)
 libname = tomcrypt
 libdir = core/lib/libtomcrypt
 include mk/lib.mk
+else ifeq ($(WITH_CRYPTO), platcrypt)
+libname = platcrypt
+libdir = core/arch/arm/plat-$(PLATFORM)/lib/libplatcrypt
```
Comment (jenswi-linaro) [inline core/core.mk]: I'm not super happy about this. How about something like:
```
CRYPTOLIB_NAME ?= tomcrypt
CRYPTOLIB_DIR ?= core/lib/libtomcrypt
libname = CRYPTOLIB_NAME
libdir = CRYPTOLIB_DIR
...
```
instead?

@jforissier do you have any input?

Comment (jforissier) [reply core/core.mk]: Agreed. Add a `CFG_` prefix and add the missing `$(...)` and that should be perfect ;-)
```
CFG_CRYPTOLIB_NAME ?= tomcrypt
CFG_CRYPTOLIB_DIR ?= core/lib/libtomcrypt
libname = $(CFG_CRYPTOLIB_NAME)
libdir = $(CFG_CRYPTOLIB_DIR)
```

@deebee-v2 would that work for your use case?

Comment (deebee-v2) [reply core/core.mk]: Yes that works perfectly, thanks!

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by jforissier** [comment]
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

(Please squash the changes into a single commit with a suitable subject, such as "Make name and path of crypto library configurable" or something like that)

**Comment by deebee-v2** [comment]
@jforissier , I'm not sure why one of the regression builds has failed, any suggestions? Or is it possible to re-start it?

**Comment by jforissier** [comment]
@deebee-v2 don't worry, the failure is unrelated to this PR. See https://github.com/linaro-swg/optee_examples/pull/22.
I will re-start the CI job as soon as the above fix is merged.

**Pull request merged**